weblate/machinery/microsoft.py

```diff
@@ -112,14 +122,15 @@ def download_languages(self):
 
     def download_translations(self, source, language, text, unit, user):
         """Download list of possible translations from a service."""
+        payload = _js.dumps([{'Text': text}])
+
         args = {
-            'text': text[:5000],
+            'api-version': '3.0',
             'from': source,
             'to': language,
-            'contentType': 'text/plain',
             'category': 'general',
         }
-        response = self.request("get", TRANSLATE_URL, params=args)
+        response = self.request("post", TRANSLATE_URL, params=args, json=payload)
```

Comment:
```suggestion
        response = self.request("post", TRANSLATE_URL, params=args, json=[{'Text': text}])
```

`json` param accepts data directly. You probably haven't tested this at all, because this could not work with the real service (in the test we do not look at request bodies, so it will pass).

Reply:
Yes i tested it and it does not work as suggested.

Reply:
Because you have there `{'Text','Hello'}`, while my suggestion is `{'Text': text}`

Reply:
But what I meant - did you test your code with real service? IMHO it does not work as it is now, because you pass it a string instead of json values.

Reply:
Yes, I try the service in shell with every commit.